comp/core/tagger/mock/mock.go

```diff
@@ -0,0 +1,149 @@
+	workloadmetafx "github.com/DataDog/datadog-agent/comp/core/workloadmeta/fx"
+	configmock "github.com/DataDog/datadog-agent/pkg/config/mock"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+// Mock implements mock-specific methods for the tagger component.
+type Mock interface {
+	tagger.Component
+
+	// SetTags allows to set tags in the mock fake tagger
+	SetTags(entityID types.EntityID, source string, low, orch, high, std []string)
+
+	// SetGlobalTags allows to set tags in store for the global entity
+	SetGlobalTags(low, orch, high, std []string)
+}
+
+// mockTaggerClient is a mock of the tagger Component
+type mockTaggerClient struct {
+	*taggerimpl.TaggerWrapper
+}
+
+// mockHandleRequest is a simple mocked http.Handler function to test the route is registered correctly on the api component
+func (m *mockTaggerClient) mockHandleRequest(w http.ResponseWriter, _ *http.Request) {
+	w.Write([]byte("OK"))
+}
+
+// New returns a Mock
+func New(t testing.TB) Mock {
+	c := configmock.New(t)
+	params := tagger.Params{
+		UseFakeTagger: true,
+	}
+	logComponent := logmock.New(t)
+	wmeta := fxutil.Test[optional.Option[workloadmeta.Component]](t,
+		fx.Provide(func() log.Component { return logComponent }),
+		fx.Provide(func() config.Component { return c }),
+		workloadmetafx.Module(workloadmeta.NewParams()),
+	)
+
+	tagger, err := taggerimpl.NewTaggerClient(params, c, wmeta, logComponent, noopTelemetry.GetCompatComponent())
+
+	assert.NoError(t, err)
+
+	return &mockTaggerClient{
+		tagger,
+	}
+}
+
+// Provides is a struct containing the mock and the endpoint
+type Provides struct {
+	fx.Out
+
+	Comp     Mock
+	Endpoint api.AgentEndpointProvider
+}
+
+type dependencies struct {
+	fx.In
+
+	Config    config.Component
+	Log       log.Component
+	WMeta     optional.Option[workloadmeta.Component]
+	Telemetry telemetry.Component
+}
+
+// NewMock returns a Provides
+func NewMock(deps dependencies) (Provides, error) {
+	params := tagger.Params{
+		UseFakeTagger: true,
+	}
+
+	tagger, err := taggerimpl.NewTaggerClient(params, deps.Config, deps.WMeta, deps.Log, deps.Telemetry)
+	if err != nil {
+		return Provides{}, err
+	}
+
+	c := &mockTaggerClient{
+		tagger,
+	}
+	return Provides{
+		Comp:     c,
+		Endpoint: api.NewAgentEndpointProvider(c.mockHandleRequest, "/tagger-list", "GET"),
+	}, nil
+}
+
+// Module is a module containing the mock, useful for testing
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(NewMock),
+		fx.Supply(config.Params{}),
+		fx.Supply(log.Params{}),
+		fx.Provide(func(t testing.TB) log.Component { return logmock.New(t) }),
+		config.MockModule(),
+		sysprobeconfigimpl.MockModule(),
+		workloadmetafx.Module(workloadmeta.NewParams()),
+		telemetryimpl.MockModule(),
+	)
+}
+
+// SetupFakeTagger calls fxutil.Test to create a mock tagger for testing
+func SetupFakeTagger(t testing.TB) Mock {
+	return fxutil.Test[Mock](t, Module())
+}
```

Comment:
How is this different from `NewMock(t)` ? Ideally we would want to not use `SetupFakeTagger()` or `Module()` as much as possible in the tests.

We might want a `NewMock(testing.TB)` method that manually instantiate all it's dependencies and a `NewMockWithDependencies(testing.TB, ...)` that take the deps in. This would produce a similar pattern that the config mock who offer `New`, `NewFromYaml`, ...